Source/KakapoServer.swift

```diff
@@ -48,16 +94,20 @@ class KakapoServer: NSURLProtocol {
         return request
     }
     
-    override class func requestIsCacheEquivalent(a: NSURLRequest, toRequest b: NSURLRequest) -> Bool {
-        return false
-    }
-    
     override func startLoading() {
         guard let requestString = request.URL?.absoluteString else { return }
+        var dataBody: NSData?
         
         for (key, object) in KakapoServer.routes {
             if let info = parseUrl(key, requestURL: requestString) {
-                object.handler(request: Request(urlString: requestString, info: info))
+                
+                if let dataFromProtocol = NSURLProtocol.propertyForKey(kkp_RequestHTTPBodyKey, inRequest: request) as? NSData {
+                    dataBody = dataFromProtocol
+                } else if let dataFromNSURLRequest = request.HTTPBody {
```

Comment:
Not deprecated, but for now since the "bug" is there the first thing will happen when using NSURLSession, whereas the second will when using old NSURLConnection